File: src/main/java/me/syari/ss/discord/api/exceptions/RateLimitedException.java
package me.syari.ss.discord.api.exceptions;

import me.syari.ss.discord.internal.requests.Route;
import org.jetbrains.annotations.NotNull;

public class RateLimitedException extends Exception {
    public RateLimitedException(@NotNull Route route, long retryAfter) {
        this(route.getBaseRoute() + ":" + route.getMajorParameters(), retryAfter);
    }

    public RateLimitedException(String route, long retryAfter) {
        super(String.format("The request was ratelimited! Retry-After: %d  Route: %s", retryAfter, route));
    }
}

File: src/main/kotlin/me/syari/ss/discord/api/exceptions/RateLimitedException.kt
package me.syari.ss.discord.api.exceptions

import me.syari.ss.discord.internal.requests.Route

class RateLimitedException(
    route: String?, retryAfter: Long
): Exception(String.format("The request was ratelimited! Retry-After: %d  Route: %s", retryAfter, route)) {
    constructor(route: Route, retryAfter: Long): this(route.baseRoute + ":" + route.majorParameters, retryAfter)
}